mmv1/third_party/terraform/services/compute/resource_compute_interconnect_group_test.go

```diff
@@ -0,0 +1,99 @@
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccComputeInterconnectGroup_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"deletion_protection": false,
+		"random_suffix":       acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInterconnectGroup_basic(context),
+			},
+			{
+				ResourceName:      "google_compute_interconnect_group.example-interconnect-group",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccComputeInterconnectGroup_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_compute_interconnect_group" "example-interconnect-group" {
+  name   = "tf-test-example-interconnect-group%{random_suffix}"
+  intent {
+    topology_capability = "NO_SLA"
+  }
+}
+`, context)
+}
+
+func TestAccComputeInterconnectGroup_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"deletion_protection": false,
+		"random_suffix":       acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckComputeInterconnectGroupDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInterconnectGroup_basic(context),
+			},
+			{
+				ResourceName:      "google_compute_interconnect_group.example-interconnect-group",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccComputeInterconnectGroup_update(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_compute_interconnect_group.example-interconnect-group", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:      "google_compute_interconnect_group.example-interconnect-group",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccComputeInterconnectGroup_update(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_compute_interconnect_group" "example-interconnect-group" {
+  name   	  = "tf-test-example-interconnect-group%{random_suffix}"
+  intent {
+    topology_capability = "NO_SLA"
+  }
+  description = "New description"
+  interconnects {
+	name = "my-interconnect"
+	interconnect = "https://www.googleapis.com/compute/v1/projects/${data.google_project.project.project_id}/global/interconnects/${google_compute_interconnect.example-interconnect.name}"
```

Comment:
It's this reference to `google_compute_interconnect.example-interconnect.name` that's causing the `Error: Reference to undeclared resource` in the update test.